esphome/components/mqtt/__init__.py

```diff
@@ -378,6 +382,9 @@ async def to_code(config):
     if CONF_CERTIFICATE_AUTHORITY in config:
         cg.add(var.set_ca_certificate(config[CONF_CERTIFICATE_AUTHORITY]))
         cg.add(var.set_skip_cert_cn_check(config[CONF_SKIP_CERT_CN_CHECK]))
+        if CONF_CERTIFICATE in config and CONF_KEY in config:
```

Comment:
With the above Inclusive change, only need to check for one item
```suggestion
        if CONF_CERTIFICATE in config:
```